src/vorta/views/partials/treemodel.py

```diff
@@ -994,3 +1042,246 @@ def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
+
+        model = self.sourceModel()
+        item = model.index(sourceRow, 0, sourceParent).internalPointer()
+
+        item_path = path_to_str(item.path)
+        item_name = item.subpath
+
+        # Exclude Parents
+        if self.searchPattern.exclude_parents:
+            if item.children:
+                return False
+
+        # Set default values
+        if self.searchPattern.match is None:
+            self.searchPattern.match = "fm" if self.searchPattern.path else "in"
+
+        if self.searchPattern.path:
+            search_item = item_path
+        else:
+            search_item = item_name
+
+        if self.searchPattern.search_string:
+            # TODO: Move operations on search string to setSearchString method
+
+            search_string = " ".join(self.searchPattern.search_string)
+
+            # Ignore Case?
+            if self.searchPattern.ignore_case:
+                search_item = search_item.lower()
+                search_string = search_string.lower()
+
+            if self.searchPattern.match == "in" and search_string not in search_item:
+                return False
+            elif self.searchPattern.match == "ex" and search_string != search_item:
+                return False
+            elif self.searchPattern.match == "re" and not re.search(search_string, search_item):
+                return False
+            elif self.searchPattern.match == "fm" and not fnmatch(search_item, search_string):
+                return False
+
+        if self.searchPattern.size:
+            # Diff view has size column corresponding to the changed_size while
+            # Extract view has size column corresponding to the size
+            if hasattr(item.data, 'changed_size'):
+                item_size = item.data.changed_size
+            else:
+                item_size = item.data.size
+
+            for filter_size in self.searchPattern.size:
+                if not compare_values_with_sign(item_size, filter_size[1], filter_size[0]):
+                    return False
+
+        if self.searchPattern.change:
+            item_change = item.data.change_type.short()
+
+            if item_change != self.searchPattern.change:
+                return False
+
+        return True
+
+
+class DiffTreeSortFilterProxyModel(FileTreeSortFilterProxyModel):
+    """
+    Tree model for diff view.
+    """
+
+    def __init__(self, parent=None) -> None:
+        super().__init__(parent)
+
+    def get_parser(self):
+        """Add Diff view specific arguments to the parser."""
+        parser = super().get_parser()
+        parser.add_argument(
+            "-b", "--balance", type=FileTreeSortFilterProxyModel.valid_size, help="Match by balance size."
+        )
+
+        return parser
+
+    def filterAcceptsRow(self, sourceRow: int, sourceParent: QModelIndex) -> bool:
+        """
+        Return whether the row should be accepted.
+        """
+
+        if not self.searchPattern:
+            return True
```

Comment:
We don't need this here, since super already implements the check.

Reply:
I added this because if super returns True (if search is empty) it will still go through child class's custom checks (balance in this case). What do you think?

Reply:
Makes sense.